src/Resolvers/DomainTenantResolver.php

```diff
@@ -28,18 +28,34 @@ class DomainTenantResolver extends Contracts\CachedTenantResolver
 
     public function resolveWithoutCache(...$args): Tenant
     {
-        /** @var Domain $domain */
-        $domain = config('tenancy.domain_model')::where('domain', $args[0])->first();
+        $domain = $args[0];
 
-        if ($domain) {
-            static::$currentDomain = $domain;
+        /** @var Tenant|null $tenant */
+        $tenant = config('tenancy.tenant_model')::query()
+            ->whereHas('domains', function ($query) use ($domain) {
+                $query->select(['tenant_id', 'domain'])->where('domain', $domain);
+            })
+            ->with([
+                'domains' => function ($query) use ($domain) {
+                    $query->where('domain', $domain);
+                },
+            ])
+            ->first();
 
-            return $domain->tenant;
+        if ($tenant) {
+            static::$currentDomain = $tenant->domains->first();
+
+            return $tenant;
         }
 
         throw new TenantCouldNotBeIdentifiedOnDomainException($args[0]);
     }
 
+    public function tenantIdentifiedFromCache(Tenant $tenant, ...$args): void
```

Comment:
This should be called from `resolveWithoutCache()` and as such renamed to something unrelated to cache. Right now this logic is duplicated between `resolveWithoutCache()` and this method.

Reply:
I just thought, that a method like `tenantIdentifiedFromCache()` makes this package more flexible, when you are writing your own resolvers and need to implement some custom logic, as name implies, when tenant is identified from cache. It's like an event (firing an event and listening for it would be an overkill and only would make more problems, for a simple task like this). Ofcourse, if it is needed, I can update code like this:
```php

public function resolveWithoutCache(...$args)
{
    // rest of the code
    if ($tenant) {
        $this->setCurrentDomain($tenant->domains->first());
    
        return $tenant;
    }

    // rest of the code
}

public function tenantIdentifiedFromCache(Tenant $tenant, ...$args)
{
    $this->setCurrentDomain($tenant);
}

protected function setCurrentDomain(Tenant $tenant)
{
    self::$currentDomain = $tenant->domains->first();
}
```

Reply:
But on the other hand, it could be simple `tenantIdentified()` method, which is called whenever Tenant is identified (from `resolve()` in *CachedTenantIdentifier* and `resolveWithoutCache()` in *DomainTenantResolver*), hmmm. Please, make final decision about this, and I will make the necessary changes.

```php
public function resolveWithoutCache(...$args)
{
    // rest of the code
    if ($tenant) {
        $this->tenantIdentified($tenant->domains->first());
    
        return $tenant;
    }

    // rest of the code
}

public function tenantIdentified(Tenant $tenant)
{
    self::$currentDomain = $tenant->domains->first();
}
```